packages/ml-natural-language/lib/index.js

```diff
@@ -89,19 +90,30 @@ class FirebaseMlKitLanguageModule extends FirebaseModule {
     );
   }
 
-  newSmartReplyConversation(messageHistoryLimit) {
-    validateOptionalNativeDependencyExists(
-      'ml_natural_language_smart_reply_model',
-      'ML Kit Smart Replies',
-      !!this.native.getSuggestedReplies,
-    );
-    if (!isUndefined(messageHistoryLimit) && !isNumber(messageHistoryLimit)) {
+  suggestReplies(messages) {
+    if (!isArray(messages)) {
       throw new Error(
-        "firebase.mlKitLanguage().newSmartReplyConversation(*) 'messageHistoryLimit' must be a number or undefined.",
+        "firebase.mlKitLanguage().suggestReplies(*) 'messages' must be an array value.",
       );
     }
 
-    return new SmartReplyConversation(this.native, messageHistoryLimit);
+    if (messages.length === 0) {
+      return Promise.resolve([]);
+    }
+
+    const validated = [];
+
+    for (let i = 0; i < messages.length; i++) {
+      try {
+        validated.push(validateTextMessage(messages[i]));
```

Comment:
Can we not just loop over the original array and validate, without pushing to another array - it'll throw anyway? If you need the index you can pass it to the validate fn.

Reply:
The validate returns a new object with defaults - I can wrap the try catch around the for?